src/main/java/frc/robot/autons/DelaySelector.java

```diff
@@ -0,0 +1,152 @@
+package frc.robot.autons;
+
+import java.util.Optional;
+
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import frc.robot.subsystems.Dashboard;
+
+public class DelaySelector {
+    // enum for delay choices
+    // s refers to seconds d refers to decimals
+    public enum DelayChoices {
+        s0,
+        s0d5,
+        s1,
+        s1d5,
+        s2,
+        s2d5,
+        s3,
+        s3d5,
+        s4,
+        s4d5,
+        s5,
+        s5d5,
+        s6,
+        s6d5,
+        s7,
+        s7d5,
+        s8,
+        s8d5,
+        s9,
+        s9d5,
+        s10,
+        s10d5,
+        s11,
+        s11d5,
+        s12,
+        s12d5,
+        s13,
+        s13d5,
+        s14,
+        s14d5;
+    }
+
+    // Choosers
+    SendableChooser<DelayChoices> delayChooser;
+
+    // Singleton Setup
+    private static DelaySelector instance;
+
+    public static DelaySelector getInstance() {
+        if (instance == null) {
+            System.out.println("Creating new DelaySelector");
+            instance = new DelaySelector();
+        }
+        return instance;
+    }
+
+    private DelaySelector() {
+        this.delayChooser = Dashboard.getInstance().getDelayChooser();
```

Comment:
instead of using a sendable chooser with a bunch of options, It would be a lot less code and a lot easier to make a selection if you just do Shuffleboard.getDouble() and then clamp it between 0 and 10.5.
That way you could do any fractional number of seconds between min and max and there isn't a long list of cases

Reply:
how do i do that ive been trying to do it like a genearic entry but it is not allowing me to go from GenericEntry to Double

Reply:
https://docs.wpilib.org/en/stable/docs/software/dashboards/shuffleboard/layouts-with-code/retrieving-data.html

Reply:
do you want me to make a new tab

Reply:
You can put it on the drivers tab. Let’s talk about this on Monday